.circleci/config.yml

```diff
@@ -9,16 +9,16 @@ version: 2.1
 parameters:
   ubuntu-2004-docker-image:
     type: string
-    # solbuildpackpusher/solidity-buildpack-deps:ubuntu2004-20
-    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:7a1e1b01eda0d1e20704279672bcfd53dbbc481898ff960958a225dea76345bd"
+    # solbuildpackpusher/solidity-buildpack-deps:ubuntu2004-21
+    default: "solbuildpackpusher/solidity-buildpack-deps@sha256:51cd059ee32128c011d2636dfb95b3baeae69bcdb1835c0440987302798d23d1"
```

Comment:
Looks good to me, but maybe the CI changes could be done in a separate PR and here we would only have the docker bump of the versions. This is at least what we did in the previous update. See https://github.com/ethereum/solidity/pull/14199 and https://github.com/ethereum/solidity/pull/14200.

Reply:
Yeah, I saw that when sifting through the PRs, but not sure why we do that? In any case, I can cherry-pick that commit into a new PR, it's not a biggie.

Reply:
My main concern here really is the omission of the osfuzz docker upgade - Kamil said we could do that, and then fix it later, since it's only for the nightly job, but still...

Reply:
> not sure why we do that?

We do that because you need to insert the hashes of the built images into CI config, but doing that triggers a rebuild of the images, which builds new images with new hashes. Adding hashes to the config must be in a separate PR.

> Kamil said we could do that, and then fix it later

TBH by later I meant just waiting a day or two to confirm it's not random CI breakage. Not omitting it from the PR. If it does not go away on its own, we need to go back to debugging it.

Reply:
> We do that because you need to insert the hashes of the built images into CI config, but doing that triggers a rebuild of the images, which gives you new hashes. 

This doesn't invalidate the old hashes though. Also, this is a major version bump, so it needs (?) to be done in the same PR, like in https://github.com/ethereum/solidity/pull/14200, which bumped evmone from v9 to v10.

> TBH by "doing it later" I meant waiting a day or two to confirm it's not random CI breakage. Not omitting it from the PR. If it does not go away on its own, we need to go back to debugging it.

Well, I guess this is going back to a draft :(

Reply:
Or we can just merge this as is, and then open a separate PR to bump to the latest hashes in the CI (produced by this PR).

Reply:
Another reason is that for the reviewer it makes it easier to verify you got the right hashes. Any change pushed here produces a new set of hashes even if you do not touch the images. It's only for the last set that you see the source in the PR. Others are a black box.

And by the way, the usual way to split it is to do *only* the images in the first PR. Then, in the PR that updates the hashes you can put any other code changes, which usually depend on having the new image (like the evmc update here).

Since I had to rebase this on `develop` for the GMP change, I did that split now - the evmc update is now in #14758.